libcxx/test/std/utilities/meta/meta.unary/meta.unary.prop/is_implicit_lifetime.pass.cpp

```diff
@@ -0,0 +1,323 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20
+
+// These compilers don't support __builtin_is_implicit_lifetime yet.
+// UNSUPPORTED: clang-17, clang-18, clang-19, gcc-14, apple-clang-16, apple-clang-17
+// XFAIL: apple-clang
```

Comment:
This `XFAIL` seems redundant with the `UNSUPPORTED` above.

Reply:
The build fails on Apple back-deployment targets, please check this:
|https://buildkite.com/llvm-project/libcxx-ci/builds/37470#0191b844-f2b4-4102-b33f-e633e18147c7
It looks as if `UNSUPPORTED: apple-clang-16, apple-clang-17` doesn't work here.
I don't know how to solve this. Any ideas?

Reply:
Would removing the `apple-clang-16, apple-clang-17` from `UNSUPPORTED` be acceptable?

Reply:
> The build fails on Apple back-deployment targets, please check this: |https://buildkite.com/llvm-project/libcxx-ci/builds/37470#0191b844-f2b4-4102-b33f-e633e18147c7 It looks as if `UNSUPPORTED: apple-clang-16, apple-clang-17` doesn't work here. Unless I misread the error message. I don't know how to solve this. Any ideas?

+1. I have the exact same question.

Reply:
I tried using UNSUPPORTED: apple-clang-18, but that seem to not have worked. Looking at the [buildkite pipeline configs](https://github.com/llvm/llvm-project/blob/main/libcxx/utils/ci/buildkite-pipeline.yml#L24), it looks like LLVM 18 runs on the apple workers.

Reply:
Nikolas helped me to solve this issue. Basically, `UNSUPPORTED: apple-clang-16, apple-clang-17` does not work because `Apple back-deployment macosx10.{13,15}` and `Apple back-deployment with hardening enabled` both run on AppleClang 15. In addition, there is no AppleClang 17. See [comment](https://github.com/llvm/llvm-project/pull/105946#discussion_r1743613080).

Reply:
@robincaloudis Thank you! I'll update the PR. In my case the builtin won't be supported even in AppleClang 17 as it is too new.

Reply:
You don't know that, so let's only add it when AppleClang17 actually exists.

Reply:
> You don't know that, so let's only add it when AppleClang17 actually exists.

I mixed the LLVM version with the AppleClang version. Sorry! Thanks!